rotkehlchen/inquirer.py

```diff
@@ -622,8 +618,38 @@ def find_curve_pool_price(
 
         # Calculate weight of each asset as the proportion of tokens value
         weights = (data[x + 1] * prices[x] / total_assets_price for x in range(len(tokens)))
-        assets_price = FVal(sum(map(operator.mul, weights, prices)))
-        return (assets_price * FVal(data[0])) / (10 ** lp_token.decimals)
+        token_weights: Dict[EthereumToken, Tuple[FVal, FVal]] = {}
+        for token, weight, price in zip(tokens, weights, prices):
+            token_weights[token] = (weight, price)
+        return token_weights
+
+    def find_curve_pool_price(
+        self,
+        lp_token: EthereumToken,
+    ) -> Optional[Price]:
+        """
+        1. Obtain the pool for this token
+        2. Obtain prices for assets in pool
+        3. Obtain the virtual price for share and the balances of each
+        token in the pool
+        4. Calc the price for a share
+
+        Returns the price of 1 LP token from the pool
+        """
+        weighted_prices = self.find_curve_pool_tokens_amount(
+            lp_token_address=lp_token.ethereum_address,
+        )
+        if weighted_prices is None:
+            return None
+        pool = CURVE_POOLS[lp_token.ethereum_address]
+        assets_price = FVal(sum([weight * price for (weight, price) in weighted_prices.values()]))
+        virtual_price_raw = self._ethereum.call_contract(  # type: ignore
+            contract_address=pool.pool_address,
+            method_name='get_virtual_price',
+            abi=CURVE_POOL_ABI,
+        )
+        virtual_price_value = FVal(virtual_price_raw)
```

Comment:
What does the contract return here? Isn't it a integer?

Reply:
Yes, it is, thanks